packages/x-tree-view/src/internals/TreeViewItemDepthContext/TreeViewItemDepthContext.ts

```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+import { TreeViewItemId } from '../../models';
+
+export const TreeViewItemDepthContext = React.createContext<
+  number | ((itemId: TreeViewItemId) => number)
```

Comment:
`SimpleTreeView` uses a number because the depth of an item is always the depth of its parent + 1 
`RichTreeView` uses a function to retrieve the depth from the state

Not sure how to improve this...